frontend/src/pages/projects/screens/detail/connections/ConnectionsDeleteModal.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import { K8sStatus } from '@openshift/dynamic-plugin-sdk-utils';
+import { Connection } from '~/concepts/connectionTypes/types';
+import DeleteModal from '~/pages/projects/components/DeleteModal';
+
+type Props = {
+  deleteConnection: Connection;
+  onClose: () => void;
+  onDelete: () => Promise<K8sStatus>;
+  refresh: () => void;
+};
+
+export const ConnectionsDeleteModal: React.FC<Props> = ({
+  deleteConnection,
+  onClose,
+  refresh,
+  onDelete,
+}) => {
+  const [isDeleting, setIsDeleting] = React.useState(false);
+  const [error, setError] = React.useState<Error>();
+
+  return (
+    <DeleteModal
+      title="Delete connection?"
+      isOpen
+      onClose={onClose}
+      submitButtonLabel="Delete"
+      onDelete={() => {
+        setIsDeleting(true);
+        setError(undefined);
+
+        onDelete()
+          .then(() => {
+            onClose();
+            refresh();
+          })
+          .catch((e) => {
+            setError(e);
+            setIsDeleting(false);
+          });
+      }}
+      deleting={isDeleting}
+      error={error}
+      deleteName={deleteConnection.metadata.annotations['openshift.io/display-name'] ?? ''}
```

Comment:
use `getDisplayNameFromK8sResource` to get the name and store it in a variable for reuse.